pkg/agent/openflow/client.go

```diff
@@ -536,11 +536,20 @@ func (c *client) InstallPodFlows(interfaceName string, podInterfaceIPs []net.IP,
 	isAntreaFlexibleIPAM := c.connectUplinkToBridge && c.nodeConfig.PodIPv4CIDR != nil && !c.nodeConfig.PodIPv4CIDR.Contains(podInterfaceIPv4)
 
 	localGatewayMAC := c.nodeConfig.GatewayConfig.MAC
-	flows := []binding.Flow{
-		c.featurePodConnectivity.podClassifierFlow(ofPort, isAntreaFlexibleIPAM),
-		c.featurePodConnectivity.l2ForwardCalcFlow(podInterfaceMAC, ofPort),
+
+	// In order to easily maintain Pod classifier flow, assign a separate cache key for it.
+	podClassifierFlow := c.featurePodConnectivity.podClassifierFlow(ofPort, isAntreaFlexibleIPAM, labelID)
+	if labelID != nil {
+		// If labelID is not nil, which means we are only trying to update Pod classifier
+		// flow with a new LabelIdentity.
+		return c.modifyFlows(c.featurePodConnectivity.podCachedFlows, fmt.Sprintf("%s/classifier", interfaceName), []binding.Flow{podClassifierFlow})
```

Comment:
this may surprise the method's consumer as setting labelID to `InstallPodFlows` means updating classifier flow only. By previous comment, I meant it could just override all flows if label ID doesn't change often. Having optimization like this is not very good for the code's maintenance. For example, the code could work with one exception like this (nil labelID means adding all flows, non-nil labelID means updating classifier folow), but would become complex and hard to understand with another exception.

Reply:
Got it. Changed.